src/main/java/com/shopping/entities/Order.java

```diff
@@ -0,0 +1,106 @@
+package com.shopping.entities;
+
+import java.time.LocalDate;
+import java.util.Set;
+import java.util.TreeSet;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "order")
+public class Order extends BaseEntity{
+	
+	@Column
+	private LocalDate ordered;
+	
+	@Column
+	private LocalDate shipped;
+	
+	@Column
+	private OrderStatus status;
+	
+	@Column
+	private Double total;
+	
+	@OneToOne
+	private Address shipTo;
+	
+	@ManyToOne
+	private Account account;
+	
+	@OneToMany
+	private Set<Payment> payments = new TreeSet<>();

```

Comment:
Just to confirm, is it better to initialize the objects in the constructor or directly?

Reply:
1. In Order we have too many fields. in constructor its not recommended to take more than 5 arguments. Hence created setter for the same.
One more thing.. if we are using setter also we may need to autowire which we are not doing now.

Hence initializing here itself looks good.